rest-server/engines/src/main/scala/com/actionml/engines/ur/URDataset.scala

```diff
@@ -60,14 +67,28 @@ class URDataset(engineId: String, val store: Store) extends Dataset[UREvent](eng
       jsonConfig,
       errorMsg = s"Error in the Algorithm part of the JSON config for engineId: $engineId, which is: " +
         s"$jsonConfig",
-      transform = _ \ "algorithm").andThen { p =>
+      transform = _ \ "algorithm"
+    ).andThen { p =>
       params = p
-
       indicatorNames = params.indicators.map(_.name)
-
       Valid(p)
+    }.andThen { p =>
+      parseAndValidate[URDatasetParams](
+        jsonConfig,
+        errorMsg = s"Error in the Dataset part pf the JSON config for engineId: $engineId, which is: " +
+          s"$jsonConfig",
+        transform = _ \ "dataset"
+      ).andThen { p =>
+        p.ttl.fold[Validated[ValidateError, _]](Valid(p)) { ttlString =>
+          Try(Duration(ttlString)).toOption.map { ttl =>
+            // We assume the DAO will check to see if this is a change and no do the reindex if not needed
+            eventsDao.createIndexes(Seq(("ttl_index", Indexed(ttl = ttl))))
```

Comment:
How is `ttl_index` mapped to a field in there stored case class? This should be a field name IMO, otherwise `ttl_index` is a magic value.

Reply:
Also the fold notation is overly hard to read.

Was there a logical error in the previous code?

Reply:
@pferrel fixed that `ttl_index` bug, this fold is new